Sources/APIAdapter+Types.swift

```diff
@@ -83,28 +85,52 @@ public enum RequestType {
     case jsonBody(Encodable)
     /// All the parameters will be sent as multipart
     /// and files too.
-    case multipart([MultipartFile])
+    case multipart([MultipartBodyPart])
     /// The parameters will be encoded using Base64 encoding
     /// and sent in request body.
     case base64Upload
 }
 
 /// Multipart file model for multipart request types.
-public struct MultipartFile: Hashable {
-    let name, filename, mimeType: String
-    let data: Data
+public struct MultipartBodyPart: Hashable {
+    let headers: [String: String]
+    let inputStream: InputStream
+    let contentLength: Int64
```

Comment:
```suggestion
    let contentLength: UInt64
```

Reply:
I wrote it like this first, but file sizes are returned as `Int64`, therefore I prefer not to cast.